### PR TITLE
test(react): add webchat component tests #BOT-31

### PR DESCRIPTION
**Pull request by AinaVendrell**
## Description
Tests for the webchat component.

## Approach taken / Explain the design
Add roles to some of the webchat components to be able to get them in the tests.

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: packages/botonic-react/tests/webchat/webchat.component.test.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        label: 'See docs',
+        url: 'https://docs.botonic.io',
+      },
+      { closeLabel: 'Close' },
+    ],
+    enableEmojiPicker: true,
+    enableAttachments: true,
+  }
+
+  window.HTMLElement.prototype.scrollTo = function () {}
```
Comment (vanbasten17) [inline packages/botonic-react/tests/webchat/webchat.component.test.jsx:21]: is this necessary?

File: packages/botonic-react/tests/webchat/webchat.component.test.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  it('Webchat has by default', async () => {
```
Comment (vanbasten17) [inline packages/botonic-react/tests/webchat/webchat.component.test.jsx:23]: webchat has (?) by default

File: packages/botonic-react/tests/webchat/webchat.component.test.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      render(<Webchat webchatHooks={result.current} />)
+    })
+    expect(screen.queryByRole('textbox')).toBeTruthy()
+    expect(screen.queryByRole('persistent-menu-icon')).toBeTruthy()
+    expect(screen.queryByRole('emoji-picker-icon')).toBeTruthy()
+    expect(screen.queryByRole('attachment-icon')).toBeTruthy()
+    expect(screen.queryByRole('send-button-icon')).toBeTruthy()
+  })
+
+  it.skip('Open EmojiPicker', async () => {
```
Comment (vanbasten17) [inline packages/botonic-react/tests/webchat/webchat.component.test.jsx:90]: for newcoming pr reviewers, we have decided to skip these tests atm and we will keep continue working on that on new tasks.

**Review by vanbasten17 — APPROVED** [review]
LGTM 👍 Remember to extract common logic into encapsulated functions, so for example for:
`expect(screen.queryByRole('persistent-menu-icon')).`
we can have a "custom" `expect` function called `toHaveRole('persistent-menu-icon')`. and use this one for check the expected results, just as a suggestion to consider in next iterations.

File: packages/botonic-react/tests/webchat/webchat.component.test.jsx (hunk 1 above)
Comment (AinaVendrell) [reply packages/botonic-react/tests/webchat/webchat.component.test.jsx:21]: Yes, the tests where the webchat is opened where failing and this solution worked. This was the error:

Captura de Pantalla 2020-10-26 a les 13 24 26

File: packages/botonic-react/tests/webchat/webchat.component.test.jsx (hunk 2 above)
Comment (AinaVendrell) [reply packages/botonic-react/tests/webchat/webchat.component.test.jsx]: I have changed the titles

**Review by dpinol — APPROVED** [review]

**Comment by dpinol** [comment]
@AinaVendrell please rebase to master

**Pull request merged**